R/get_entity_ids.R

```diff
@@ -0,0 +1,54 @@
+#' Get the entity info data from the FactSet database and prepare the
+#' `factset_entity_info` tibble
+#'
+#' @param conn database connection
+#'
+#' @return A tibble properly prepared to be saved as the
+#'   `factset_entity_info.rds` output file
+#'
+#' @export
+
+get_entity_ids <- function(conn, colname = "factset_entity_id") {
+
+  logger::log_debug("Extracting all values of ", colname, " from database")
+
+  logger::log_trace("Identifying tables with column: ", colname, ".")
+  table_has_entity_id_col <- vapply(
+    X = DBI::dbListTables(conn),
+    FUN = function(x) {
+      any(DBI::dbListFields(conn, x) == colname)
+    },
+    FUN.VALUE = logical(1L),
+    USE.NAMES = TRUE
+  )
+  tables_to_extract <- names(table_has_entity_id_col)[table_has_entity_id_col]
+  logger::log_trace("Tables with column: ", colname, ": ",
+    paste(tables_to_extract, collapse = ", ")
```

Comment:
NIT:
```suggestion
    toString(tables_to_extract)
```

is more compact e.g.
``` r
toString(letters[1:3])
#> [1] "a, b, c"
```